File: homework/hw1-lushenkov/internal/domain/animal/wolf.go
package animal

type Wolf struct{ *Predator }

func NewWolf(name string, foodPerDayKg int) (*Wolf, error) {
	p, err := NewPredator(name, "Wolf", foodPerDayKg)
	if err != nil {
		return nil, err
	}
	return &Wolf{Predator: p}, nil
}

File: homework/src/main/java/hse/kpo/homework1/living/Wolf.java
package hse.kpo.homework1.living;

import java.util.Map;

/**
 * Конкретный класс для волков (хищное животное).
 * Следует принципу Liskov Substitution Principle (LSP):
 * может использоваться везде, где ожидается Predator или Animal.
 */
public class Wolf extends Predator {
    public Wolf(String name, int food, int number) {
        super(name, food, number);
    }

    public Wolf(Map<String, Object> params) {
        this(
            (String) params.get("name"),
            (Integer) params.get("food"),
            (Integer) params.get("inventoryNumber")
        );
    }
}
